### PR TITLE
Skip last checkout screen when there's only one payment option available

### PR DESCRIPTION
**Pull request by atavism**
Resolves https://github.com/getlantern/engineering/issues/1366

### CODE REVIEW

**Comment by atavism** [comment]
Looks like the TestProxying test is failing..

```
    android_test.go:179:
        	Error Trace:	/Users/atavism/go/src/github.com/getlantern/lantern-client/internalsdk/android_test.go:179
        	            				/Users/atavism/go/src/github.com/getlantern/lantern-client/internalsdk/android_test.go:96
        	Error:      	Received unexpected error:
        	            	failed to WebSocket dial: failed to send handshake request: Get "https://relay.lantern.io/allocate": context deadline exceeded
        	Test:       	TestProxying
        	Messages:   	should allocate relay
```

**Comment by atavism** [comment]
I'm not sure what the status of replay.lantern.io is. But since messaging is disabled, I'm just gonna disable the tests as well.

File: lib/plans/plan_details.dart [hunk 1]
```diff
@@ -142,15 +156,34 @@ class PlanCard extends StatelessWidget {
...
-        Checkout(
-          plan: plan,
-          isPro: isPro,
-        ),
-      );
+    } else if (isDesktop()) {
+      final paymentMethods = await sessionModel.paymentMethodsv4();
+      final providers = paymentProvidersFromMethods(paymentMethods);
+      // if only one payment provider is returned, bypass the last checkout screen
+      if (providers.length == 1) {
```
Comment (jigar-f) [inline lib/plans/plan_details.dart:163]: @atavism Can we make sure that if the provider is only one and it not stripe then it should still goes to next screen


 ```
 if (providers.length == 1 && providers[0].name.toPaymentEnum() == Providers.stripe) {
        final provider = providers[0];
        final providerName = provider.name.toPaymentEnum();
        // Note: as of now, Stripe is the only payment provider that collects email

        final redirectUrl = await sessionModel.paymentRedirectForDesktop(
            context,
            plan.id,
            "",
            providerName,
          );
          await openDesktopWebview(
              context: context,
              provider: providerName,
              redirectUrl: redirectUrl);
          return;
      }
```

**Review by jigar-f — COMMENTED** [review]
@atavism Just one comment

File: lib/plans/plan_details.dart (hunk 1 above)
Comment (atavism) [reply lib/plans/plan_details.dart:163]: > Can we make sure that if the provider is only one and it not stripe then it should still goes to next screen

@jigar-f Yes, that's how it currently works, we are just checking if the payment provider is Stripe further down:

```kotlin
      if (providers.length == 1) {
        final provider = providers[0];
        final providerName = provider.name.toPaymentEnum();
        // Note: as of now, Stripe is the only payment provider that collects email
        if (providerName == Providers.stripe) {
          final redirectUrl = await sessionModel.paymentRedirectForDesktop(
            context,
            plan.id,
            "",
            providerName,
          );
          await openDesktopWebview(
              context: context, provider: providerName, redirectUrl: redirectUrl);
          return;
        }
      }
```

I made those changes and added a comment to make it more explicit Stripe is the only payment provider this applies to

**Comment by atavism** [comment]
Thanks for the review, @jigar-f! Just made the changes you requested. Does it look good to you now?

**Comment by jigar-f** [comment]
Looks good @atavism, Merging.

**Pull request merged**